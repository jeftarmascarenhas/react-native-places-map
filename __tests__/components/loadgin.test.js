import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Loading from '../../src/components/loading'

describe('<Loading />', () => {
  it('rendering', () => {
    const wrapper = shallow(<Loading />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.children()).toHaveLength(2)
  })
  it('should have title when passed title return Loading or Aguarde', () => {
    const wrapper = shallow(<Loading title="Loading" />)
    expect(wrapper.props().children[1].props.children).toEqual('Loading')
    const wrapperB = shallow(<Loading title="Aguarde" />)
    expect(wrapperB.props().children[1].props.children).toEqual('Aguarde')
  })
  it('should not have title when title not passed title return Loading...', () => {
    const wrapper = shallow(<Loading />)
    expect(wrapper.props().children[1].props.children).toEqual('Loading...')
  })
})
