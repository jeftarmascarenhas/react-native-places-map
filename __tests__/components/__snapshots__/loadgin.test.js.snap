// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> rendering 1`] = `
<Styled(View)>
  <Styled(ActivityIndicator) />
  <Styled(Text)>
    Loading...
  </Styled(Text)>
</Styled(View)>
`;
